src/backend/tty.rs

```diff
@@ -257,6 +257,33 @@ struct ConnectorProperties<'a> {
     connector: connector::Handle,
     properties: Vec<(property::Info, property::RawValue)>,
 }
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+struct PciID {
+    device_id: String,
+}
+
+impl PciID {
+    fn from_drm_node(node: &DrmNode) -> Option<PciID> {
+        use std::fs;
+
+        let device_id = fs::read_to_string(format!(
+            "/sys/dev/char/{}:{}/device/device",
+            DrmNode::major(node),
+            DrmNode::minor(node)
+        ));
+
+        match device_id {
+            Ok(device_id) => Some(PciID { device_id }),
+            Err(err) => {
+                error!(
```

Comment:
error! is for programming errors only: https://yalter.github.io/niri/Development%3A-Developing-niri.html#logging-levels